src/util/taproot.rs

```diff
@@ -169,4 +920,105 @@ mod test {
+    }
+
+    #[test]
+    fn build_huffman_tree() {
+        let secp = Secp256k1::verification_only();
+        let internal_key = schnorr::PublicKey::from_str("93c7378d96518a75448821c4f7c8f4bae7ce60f804d03d1f0628dd5dd0f5de51").unwrap();
+
+        let script_odds = vec![
+            (10, Script::from_hex("51").unwrap()), // semantics of script don't matter for this test
+            (20, Script::from_hex("52").unwrap()),
+            (20, Script::from_hex("53").unwrap()),
+            (30, Script::from_hex("54").unwrap()),
+            (20, Script::from_hex("55").unwrap()),
+        ];
+        let tree_info = TaprootSpendInfo::new_huffman_tree(&secp, internal_key, script_odds.clone()).unwrap();
+
+        // Obtain the output key
+        let output_key = tree_info.output_key();
+
+        // Try to create and verify a control block from each path
+        for (_odds, script) in script_odds {
+            let ver_script = (script, LeafVersion::default());
+            let ctrl_block = tree_info.control_block(&ver_script).unwrap();
+            assert!(ctrl_block.verify_taproot_commitment(&secp, &output_key, &ver_script.0))
+        }
+    }
+
+    #[test]
+    fn taptree_builder() {
+        let secp = Secp256k1::verification_only();
+        let internal_key = schnorr::PublicKey::from_str("93c7378d96518a75448821c4f7c8f4bae7ce60f804d03d1f0628dd5dd0f5de51").unwrap();
+
+        let builder = TaprootBuilder::new();
+        // Create a tree as shown below
+        // For example, imagine this tree:
+        // A, B , C are at depth 2 and D,E are at 3
+        //                                       ....
+        //                                     /      \
+        //                                    /\      /\
+        //                                   /  \    /  \
+        //                                  A    B  C  / \
+        //                                            D   E
+        let a = Script::from_hex("51").unwrap();
+        let b = Script::from_hex("52").unwrap();
+        let c = Script::from_hex("53").unwrap();
+        let d = Script::from_hex("54").unwrap();
+        let e = Script::from_hex("55").unwrap();
+        let builder = builder.add_leaf(2, a.clone()).unwrap();
+        let builder = builder.add_leaf(2, b.clone()).unwrap();
+        let builder = builder.add_leaf(2, c.clone()).unwrap();
+        let builder = builder.add_leaf(3, d.clone()).unwrap();
+        let builder = builder.add_leaf(3, e.clone()).unwrap();
+
+        let tree_info = builder.finalize(&secp, internal_key).unwrap();
+        let output_key = tree_info.output_key();
+
+        for script in vec![a, b, c, d, e] {
+            let ver_script = (script, LeafVersion::default());
+            let ctrl_block = tree_info.control_block(&ver_script).unwrap();
+            assert!(ctrl_block.verify_taproot_commitment(&secp, &output_key, &ver_script.0))
+        }
+    }
```

Comment:
I mean in the case where there are no "real" scripts, but only the unspendable script suggested by [BIP 341](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#constructing-and-spending-taproot-outputs):

> If the spending conditions do not require a script path, the output key should commit to an unspendable script path instead of having no script path. This can be achieved by computing the output key point as Q = P + int(hashTapTweak(bytes(P)))G

Not sure how opinionated this library should be about that recommendation.